src/main/java/org/apache/sysml/hops/FunctionOp.java

```diff
@@ -55,6 +59,8 @@
 	private String[] _outputs = null; 
 	private ArrayList<Hop> _outputHops = null;
 	private boolean _singleOutFun = false;
+
+	private boolean _fopRecompileOnce = false;
```

Comment:
as I said above, let's not focus on IPA or recompile once in this PR - we can view them as optimizations that will follow subsequently.